src/librbd/exclusive_lock/PreReleaseRequest.cc

```diff
@@ -193,10 +190,7 @@ void PreReleaseRequest<I>::handle_invalidate_cache(int r) {
   } else if (r < 0 && r != -EBUSY) {
     lderr(cct) << "failed to invalidate cache: " << cpp_strerror(r)
                << dendl;
-    m_image_ctx.io_work_queue->unblock_writes();
```

Comment:
Just add ```-EIO``` to the ignore list in the conditional

Reply:
We may add -EIO or -ETimeout to the ignore. But different from -EBLACKLISTED, we should consider these cases as failed, so we must save_result(r), so maybe we can add another case, how about that?

Reply:
Sorry, I meant ```-ETIMEDOUT``` vs ```-EIO```. Why do you need to save the error code? A failed IO doesn't really have anything to do w/ the lock failing to be released. No matter what, since you set a time out, you can expect your block device to be corrupt if using a writeback cache.